cips/cip-039.md

```diff
@@ -0,0 +1,99 @@
+
+To disable the token filter for Hyperlane, the WarpKeeper configuration MUST be modified to include synthetic token support by adding the `HYP_TOKEN_TYPE_SYNTHETIC` token type to the allowed token types array.
+
+**Current Configuration:**
+
+```go
+app.WarpKeeper = warpkeeper.NewKeeper(
+    encodingConfig.Codec,
+    encodingConfig.AddressCodec,
+    runtime.NewKVStoreService(keys[warptypes.ModuleName]),
+    govModuleAddr,
+    app.BankKeeper,
+    &app.HyperlaneKeeper,
+    []int32{int32(warptypes.HYP_TOKEN_TYPE_COLLATERAL)}, // Only collateral tokens
+)
+```
+
+**Updated Configuration:**
+
+```go
+app.WarpKeeper = warpkeeper.NewKeeper(
+    encodingConfig.Codec,
+    encodingConfig.AddressCodec,
+    runtime.NewKVStoreService(keys[warptypes.ModuleName]),
+    govModuleAddr,
+    app.BankKeeper,
+    &app.HyperlaneKeeper,
+    []int32{int32(warptypes.HYP_TOKEN_TYPE_COLLATERAL), int32(warptypes.HYP_TOKEN_TYPE_SYNTHETIC)}, // Add synthetic tokens
+)
+```
+
+To have similar functionality for IBC, the IBC token filter that prevents non-TIA tokens from being transferred via IBC MUST be removed to allow non-TIA assets to flow through IBC channels.
+
+This token filter implementation that currently exists in [`app/app.go` at line 357](https://github.com/celestiaorg/celestia-app/blob/c1f2a4c5c773f20c25043f98f4b8759b603ce825/app/app.go#L357).
+
+## Parameters
+
+This CIP does not introduce new parameters but modifies the existing Hyperlane and IBC implementations to disable the token filter.
+
+## Rationale
+
+The implementation requires only small modifications to the WarpKeeper configuration and removal of the IBC token filter, minimizing the risk of introducing bugs.
+
+## Backwards Compatibility
+
+This CIP is backward compatible. Existing functionality with moving around TIA will remain unchanged. The addition of non-TIA assets is purely additive and does not affect existing users or applications.
+
+No existing APIs, interfaces, or behaviors are modified - only new functionality is added.
+
+## Test Cases
+
+- An E2E test that transfers a non-TIA asset via IBC from an IBC-enabled chain like Osmosis to Celestia Transfer. Verify the token appears in Celestia's state machine. Confirm token can be transferred via IBC to the same chain and also another IBC-enabled chain.
+- The same test as above with Hyperlane from/to a Hyperlane-enabled chain.
+
+### Backward Compatibility
+
+- Verify TIA transfers continue to work as expected
+- Confirm no regression in existing Hyperlane and IBC functionality
+
+## Security Considerations
+
+Removing the token filter for Hyperlane and IBC introduces several security risks that must be carefully managed:
+
+### Primary Risk Categories
+
+**Malicious Token Introduction**: Arbitrary tokens can be bridged to Celestia, enabling spam tokens, malicious contracts, and potential rug pulls. Attackers could create worthless tokens to clog the network or deceive users with misleading metadata.
```

Comment:
why malicious contracts? 

I would think that spam tokens falls under resource consumption and there is only one category rather than 2. Else could you explain the actual way it can be misleading.